lib/ignore/ignore.go

```diff
@@ -214,6 +214,9 @@ func (m *Matcher) parseLocked(r io.Reader, file string) error {
 // Match matches the patterns plus temporary and internal files.
 func (m *Matcher) Match(file string) (result ignoreresult.R) {
 	switch {
+	case !utf8.ValidString(file):
```

Comment:
Lets not do this here - except for `lib/fs` that you are also fixing, the input will always be valid utf-8.

Reply:
It seems not, reverting this commit leads to the same panic originating from `watchLoop` when files are added/removed:
https://github.com/syncthing/syncthing/blob/45beb28fa5b051d14a10f102a05ec46d36ecaa75/lib/fs/basicfs_watch.go#L93
I'll push another commit to address it